TipCatDotNet.Api/Services/Payments/PaymentService.cs

```diff
@@ -70,6 +68,14 @@ async Task<Result<PaymentIntent>> ProceedPayment()
                     return Result.Failure<PaymentIntent>(ex.Message);
                 }
             }
+
+
+            long CalculateAmount(decimal amount)
+            {
+                var precision = (Decimal.GetBits(amount)[3] >> 16) & 0x000000FF;
+                var result = Convert.ToSingle(amount) * Math.Pow(10,precision);
+                return Convert.ToInt64(result);
+            }
```

Comment:
Good, but let's keep it platform indifferent 
```csharp
long ToFractionalUnits(in MoneyAmount amount)
    => (long) amount.Amount * (decimal)Math.Pow(10, amount.Currency.GetDecimalDigitsCount());
```

Reply:
Thank you, it looks much better)